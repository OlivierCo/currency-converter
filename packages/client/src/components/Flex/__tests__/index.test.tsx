import React from 'react'
import { shallow, mount } from 'enzyme'
import { FlexBox } from '../index'

describe('FlexBox', () => {
  it('should be defined', () => {
    expect(FlexBox).toBeDefined()
  })

  it('should render correctly', () => {
    const wrapper = shallow(
      <FlexBox>
        <div>test</div>
      </FlexBox>,
    )

    expect(wrapper).toMatchSnapshot()
  })

  it('should get props', () => {
    const wrapper = mount(
      <FlexBox column="true" justify="center" align="center">
        <div>test</div>
      </FlexBox>,
    )

    expect(wrapper.prop('column')).toEqual('true')
    expect(wrapper.prop('justify')).toEqual('center')
    expect(wrapper.prop('align')).toEqual('center')
  })
})
