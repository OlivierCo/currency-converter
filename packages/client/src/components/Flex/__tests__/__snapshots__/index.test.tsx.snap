// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlexBox should render correctly 1`] = `
<EmotionCssPropInternal
  __EMOTION_LABEL_PLEASE_DO_NOT_USE__="FlexBox"
  __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
  css={
    Object {
      "map": undefined,
      "name": "kdgkbl",
      "next": undefined,
      "styles": "display:flex;flex:1;flex-direction:row;justify-content:flex-start;;align-items:flex-start;;",
      "toString": [Function],
    }
  }
>
  <div>
    test
  </div>
</EmotionCssPropInternal>
`;
